src/tests/ftest/io/nvme_pool_capacity.py

```diff
@@ -0,0 +1,268 @@
+            # Create the IOR threads
+            threads = []
+            for val in range(0, num_pool):
+                pool[val] = TestPool(self.context,
+                                     dmg_command=self.get_dmg_command())
+                pool[val].get_params(self)
+                # Split total SCM and NVME size for creating multiple pools.
+                pool[val].scm_size.value = int(float(test[0])) / num_pool
+                pool[val].nvme_size.value = int(float(test[1])) / num_pool
+                pool[val].create()
+                display_string = "pool{} space at the Beginning".format(val)
+                self.pool = pool[val]
+                self.pool.display_pool_daos_space(display_string)
+
+                for thrd in range(0, num_jobs):
+                    # Based on pools/jobs, split block size
+                    # if thrd == 0:
+                    #    tmp = int(float(test[3]) / num_pool)
+                    #    test[3] = str(tmp)
+                    # Add a thread for these IOR arguments
+                    threads.append(threading.Thread(target=self.ior_thread,
+                                                    kwargs={"pool": pool[val],
+                                                            "oclass": oclass,
+                                                            "api": api,
+                                                            "test": test,
+                                                            "flags": flags,
+                                                            "results":
+                                                            self.out_queue}))
+            # Launch the IOR threads
+            for thrd in threads:
+                self.log.info("Thread : %s", thrd)
+                thrd.start()
+                time.sleep(5)
+            # Wait to finish the threads
+            for thrd in threads:
+                thrd.join()
+
+            # Verify the queue and make sure no FAIL for any IOR run
+            # Test should fail with ENOSPC.
+            while not self.out_queue.empty():
+                if (self.out_queue.get() == "FAIL" and test[4] == "PASS") \
+                     or (self.out_queue.get() != "FAIL" and test[4] == "FAIL"):
+                    self.fail("FAIL")
+
+            for val in range(0, num_pool):
+                display_string = "Pool{} space at the End".format(val)
+                self.pool = pool[val]
+                self.pool.display_pool_daos_space(display_string)
+                self.pool.destroy()
+
+    def test_nvme_pool_capacity(self):
+        """Jira ID: DAOS-2085.
+
+        Test Description:
+            Purpose of this test is to verify whether DAOS stack
+            report NOSPC when accessing data beyond pool size.
+            Use Cases
+            Test Case 1:
+             1. Perform IO less than entire SSD disk space.
+            Test Case 2:
```

Comment:
If we uncomment the blocked test cases, does this code run test case 2? If so, I think we should modify the log to say "1 or 2" on L259 and L263 because it looks like this code only runs test case 1 and 3.

Reply:
I am not sure whether we need 1 or 2 comment.... At some point, test case 2 will be done once DAOS-4803 is addressed. Presently, we found a bug with DAOS. I don't think we should fix the comment for based on existing bug.